frontend/src/components/common/form/Typeahead.tsx

```diff
@@ -61,7 +61,7 @@ export function TypeaheadField<T extends TypeaheadModel>({
         isValid={!filter && isValid}
         selected={getOptionByValue(getIn(values, name))}
         onChange={(newValues: T[]) => {
-          setFieldValue(name, getIn(newValues[0], 'value') ?? newValues[0]);
+          setFieldValue(name, getIn(newValues[0], 'name') ?? newValues[0]);
```

Comment:
does this work? I think the api expects the agency id not the agency name to be sent over the wire.

Reply:
The basic tests seem to work.  I was able to filter.  But I'm worried this isn't right, as I really don't feel comfortable with my knowledge in this area.

Reply:
Probably the safest way then would be to create a prop that takes onChange as a function, and then you can override that for the specific case that isn't working:

onChange={
          onChange
            ? onChange
            : (newValues: T[]) => {
                setFieldValue(name, getIn(newValues[0], 'value') ?? newValues[0]);
              }
        }